test/apps/iodemo/io_demo.cc

```diff
@@ -93,19 +94,29 @@ class P2pDemoCommon : public UcxContext {
     };
 
     P2pDemoCommon(const options_t& test_opts) :
-        UcxContext(test_opts.iomsg_size), _test_opts(test_opts) {
-
-        _data_buffer.resize(opts().max_data_size + ALIGNMENT);
-        uintptr_t ptr = (uintptr_t)&_data_buffer[0];
-        _padding = ((ptr + ALIGNMENT - 1) & ~(ALIGNMENT - 1)) - ptr;
+        UcxContext(test_opts.iomsg_size), _test_opts(test_opts),
+        _cur_buffer_idx(0) {
+
+        _data_buffers.resize(opts().num_buffers);
+        for (size_t i = 0; i < _data_buffers.size(); ++i) {
+            std::string &data_buffer = _data_buffers[i];
+            data_buffer.resize(opts().max_data_size + ALIGNMENT);
+            uintptr_t ptr = (uintptr_t)&data_buffer[0];
+            _padding = ((ptr + ALIGNMENT - 1) & ~(ALIGNMENT - 1)) - ptr;
+        }
     }
 
     const options_t& opts() const {
         return _test_opts;
     }
 
-    void *buffer() {
-        return &_data_buffer[_padding];
+    inline void *buffer() {
+        return &_data_buffers[_cur_buffer_idx][_padding];
+    }
+
+    inline void *next_buffer() {
+        _cur_buffer_idx = (_cur_buffer_idx + 1) % _data_buffers.size();
+        return buffer();
```

Comment:
no need to return anything, i see return value is unused

Reply:
fixed